### PR TITLE
Extend yast2_software_management.pm

### PR DESCRIPTION
**Pull request by Zaoliang**
- add basic checks for search, pattern, show list and summary
- please also see ticket for details:
  https://progress.opensuse.org/issues/20966
- reference test: 
  http://e13.suse.de/tests/3919#step/yast2_software_management
- please also check needles PR: 
  https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/249

### CODE REVIEW

File: tests/yast2_gui/yast2_software_management.pm [hunk 1]
```diff
@@ -8,11 +8,11 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without any warranty.
 
-# G-Summary: Add YaST2 UI tests
+# Summary: Add YaST2 UI tests
```
Comment (okurz) [inline tests/yast2_gui/yast2_software_management.pm]: Please improve summary

Comment (Zaoliang) [inline tests/yast2_gui/yast2_software_management.pm]: okay, change this

**Comment by okurz** [comment]
Please tidy up your commits

**Comment by Zaoliang** [comment]
@okurz Summary text got changed now, it should be fine now, please have a look at it

**Comment by okurz** [comment]
Please update your reference run after the needle update so that we can check the matching areas. Also I recommend to run the test modules multiple times to ensure no instability is increased.

Also keep in mind that still new tests are lower priority, especially for yast2_ui. I thought we were clear on that yast2_ui are remotely trusted by RMs so please make sure tests are stable there

**Comment by Zaoliang** [comment]
I thought we've agreed already on this. I extend this for TW only atm, right? This has nothing to do with instable yast2_ui tests, because it has been working quite stable for TW since a couple of improvements...

**Pull request closed without merging**